### DISCUSSION TITLE
How to implement a c++ version of glb decoder and export uncompressed glb files

### DISCUSSION THREAD

**Discussion by xinghaoyujianni** [zeux/meshoptimizer#647] [Q&A]
Hopefully a c++ version of the decoder can be implemented to decode compressed glb files to uncompressed glb for use by other c++3D programs

**Comment by zeux** [comment]
Are you looking for a C++ code example? If so, if you are using cgltf you can use the snippet here: https://github.com/jkuhlmann/cgltf/pull/129#issue-739550034 to decompress the buffers.

If you're looking for command line GLB conversion, I plan to implement support for EXT_meshopt_compression in gltfpack in input files in the future. It's also worth noting that gltf-transform supports EXT_meshopt_compression out of the box, so you can `gltf-transform cp compressed.glb result.glb` to decompress if you need to.
